lib/network/search_service.dart

```diff
@@ -0,0 +1,160 @@
+// search_service.dart
+import 'dart:async';
+
+import 'package:graphql_flutter/graphql_flutter.dart';
+import '../models/enums/search_event_type.dart';
+import '../models/server_sent_event.dart';
+import '../network/api_client.dart';
+import '../screens/chat.dart';
+import 'authentication_service.dart';
+
+const String startPersonalWishlistMutations = r'''
+  mutation startPersonalWishlist($dto: WishlistCreateDtoInput!) {
+    startPersonalWishlist(dto: $dto) {
+      createdById, id, name, type
+    }
+  }
+''';
+
+SearchEventType type = SearchEventType.message;
+
+class SearchService {
+  final ApiClient client = ApiClient();
+
+  late final _sseController = StreamController<ServerSentEvent>();
+
+  Stream<ServerSentEvent> get sseStream => _sseController.stream;
+
+  bool checkerForProduct() {
+    return type == SearchEventType.product;
+  }
+
+  bool checkerForSuggestion() {
+    return type == SearchEventType.product;
+  }
+
+  String? wishlistId;
+
+  Future<String?> generateNameForPersonalWishlist(String wishlistId) async {
+    final options = MutationOptions(
+      document: gql('''
+      mutation GenerateNameForPersonalWishlist(\$wishlistId: String!) {
+        generateNameForPersonalWishlist(wishlistId: \$wishlistId) {
+          id
+          name
+        }
+      }
+    '''),
+      variables: {'wishlistId': wishlistId},
+    );
+
+    final result = await client.mutate(options);
+
+    if (result != null && result.containsKey('generateNameForPersonalWishlist')) {
+      final name = result['generateNameForPersonalWishlist']['name'];
+      return name;
+    }
+
+    return null;
+  }
+
+  Future<String> startPersonalWishlist(String message) async {
+
+    // Перевіряємо, чи вже створений wishlist
+    if (wishlistId == null) {
+      final options = MutationOptions(
+        document: gql(startPersonalWishlistMutations),
+        variables: <String, dynamic>{
+          'dto': {'firstMessageText': message, 'type': 'Product'},
+        },
+      );
+
+      final result = await client.mutate(options);
+
+      if (result != null && result.containsKey('startPersonalWishlist')) {
```

Comment:
if it fails we cannot proceed, so we should display some kind of an error to user and have a way to re-try

`Future enhancement`